### DISCUSSION TITLE
stand-alone python code to capture button press events failing

### DISCUSSION THREAD

**Discussion by jcoe2412** [zigpy/zigpy#1469] [Q&A]
Dears

I have a SONOFF zigbee button paired with my Aeotec Zigbee dongle. While I do capture the button presses in HomeAssistant, I don't succeed to do so with my own stand-alone sample code, can someone tell me what is wrong with the code hereunder ? Nothing is being printed, so nothing is being captured.

from bellows.zigbee.application import ControllerApplication
from zigpy.zcl.clusters.general import OnOff

# Initialize the Zigbee application
app = ControllerApplication(({
        "device": {
            "path": "/dev/serial/by-id/usb-1a86_USB_Serial-if00-port0",
			'baudrate': 115200,
			'flow_control': 'software'
        }
    }))

# Function to handle button press events
def button_press_handler(cluster, command, args):
    print(f"Button pressed! Cluster: {cluster}, Command: {command}, Args: {args}")
    

# Add the button press handler to the OnOff cluster
app.add_listener(button_press_handler)

while (1) :
   time.sleep(0.1)

**Comment by puddly** [comment]
Please take a look at https://github.com/zigpy/zha. Zigpy is a low-level library and does not set up bindings and attribute reporting on its own.

**Comment by dmulcahey** [comment]
The listener needs to be added to the specific output cluster on the endpoint on the device that is generating the events… not the controller application

**Reply by dmulcahey** [reply]
https://github.com/zigpy/zha/blob/f2e1a0046881ba85187b708a27ea25b63fb5fe1a/zha/zigbee/cluster_handlers/__init__.py#L697

**Comment by jcoe2412** [comment]
thanks, being a novice here I actually started from a python code generated by Copilot/ChatGPT and the following : https://github.com/zigpy/zigpy/issues/452  I was hoping to find a turnkey solution like Copilot is often giving me, I guess I'll need to spend more time studying all this. Originally I was hoping to find a simple command line sniffer that I could run and just capture the output from my python script, but to my surprise I didn't found any of such simple sniffer. I could of course install zigbee to MQTT, but that would be overkilled here as I want to spare my raspberry PI CPU for other purposes than a running a docker image to host a mosquito server. I did pair by button with my dongle with homeassistant on a spare raspberry, and I am not planning to need anything else than capturing the button press event now that the dongle is installed on my production device. My initial naive idea was that I could capture the serial communication on ttyUSB0 and simply identify manually the data pattern linked to the button press event, but that didn't work neither, I don't succeed to sniff the data flow with minicom nor screen.
